File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.cpp
#include <string>
#include <vector>
#include <queue>
#include <algorithm>
#include <iostream>
using namespace std;

vector<int> solution(int k, vector<int> score) 
{
    vector<int> answer;
    vector<int> reward;
    int sSize = score.size();
    
    for(int i=0;i<sSize;++i)
    {
        reward.emplace_back(score[i]);
        sort(reward.begin(), reward.end(),greater<int>());
        
        if(i+1 < k)
        {
            answer.emplace_back(reward[i]);
        }else
        {
            answer.emplace_back(reward[k-1]);
        }
        
    }
    
    
    
    
    
    
    
    return answer;
}

File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.cs
using System;
using System.Collections.Generic;
public class Solution {
    public int[] solution(int k, int[] score) 
    {
        int[] answer = new int[score.Length];
        int i=0;
        List<int> reward = new List<int>(k);
        
        
        for(;i<k&&i<score.Length;++i)
        {
            reward.Add(score[i]);
            reward.Sort( (x,y) => y-x );
            answer[i] = reward[reward.Count-1];
        }
        
        for(;i<score.Length;++i)
        {
            if(score[i] > reward[reward.Count-1])
            {
                reward.RemoveAt(reward.Count-1);
                reward.Add(score[i]);
            }
            reward.Sort((x,y) => y-x);
            answer[i] = reward[reward.Count-1];
        }
        
        
                
        return answer;
    }
}